pyuoi/resampling.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+from collections import Counter
+from sklearn.model_selection import train_test_split
+from sklearn.utils import check_random_state
+
+
+def resample(type, x, replace=True, random_state=None, **kwargs):
+    r"""Takes the data in X and y and returns the appropriate
+    resampled versions.
+
+    Parameters
+    ----------
+    type : str "bootstrap" | "block"
+        type of resampling to do
+    x : array-like with shape (n_samples,)
+        A suitable array whose indices will be used as the basis for resampling.
+    replace : bool
+        Should we sample with replacement or not? True corresponds to a
+        bootstrap, False to use sklearn train_test_split
+    random_state : int or np.random.RandomState instance
+        If an int or None, used to create a np.random.RandomState instance.
+        Else, if already an instance of RandomState, used as-is to generate
+        random samples
+    **kwargs : arguments necessary to perform the desired resampling.
+    See functions below for necessary keyword arguments
+
+    Returns
+    -------
+    train_idxs: ndarray, shape (n_train_samples, n_features)
+    test_idxs: ndarray, shape (n_test_samples, n_features)
+    """
+    random_state = check_random_state(random_state)
+
+    if type == 'bootstrap':
+        train_idxs, test_idxs = bootstrap(x, random_state, replace, **kwargs)
+    elif type == 'block':
+        train_idxs, test_idxs = block_bootstrap(x, random_state, **kwargs)
+
+    return [train_idxs, test_idxs]
+
+
+def bootstrap(x, rand_state, replace, sampling_frac=0.9, stratify=None):
```

Comment:
nitpick, but can you do `rand_state` -> `random_state` in this function and `block_bootstrap` to make things consistent.